termuic/termuiConsole.go

```diff
@@ -0,0 +1,214 @@
+	sigTerm := make(chan os.Signal, 2)
+	signal.Notify(sigTerm, os.Interrupt, syscall.SIGTERM)
+
+	drawTicker := time.NewTicker(refreshInterval).C
+
+	for {
+		select {
+		case <-drawTicker:
+			tc.refreshDataForConsole()
+
+		case <-sigTerm:
+			ui.Clear()
+
+			return
+		case e := <-uiEvents:
+			tc.processUiEvents(e)
+		}
+	}
+}
+
+func (tc *TermuiConsole) processUiEvents(e ui.Event) {
+	switch e.ID {
+	case "<Resize>":
+		payload := e.Payload.(ui.Resize)
+		tc.grid.SetRect(0, 0, payload.Width, payload.Height)
+		ui.Clear()
+		ui.Render(tc.grid)
+	case "q":
+		ui.Clear()
+		tc.grid = tc.changeConsoleDisplay()
+
+	case "<C-c>":
+		ui.Close()
+		if p, err := os.FindProcess(os.Getpid()); err != nil {
+			return
+		} else {
+			if runtime.GOOS == "windows" {
+				_ = p.Kill()
+			} else {
+				_ = p.Signal(syscall.SIGINT)
+			}
+		}
+		return
+	}
+}
+
+func (tc *TermuiConsole) refreshDataForConsole() {
+	if tc.numLogLines == 10 {
+		tc.prepareDataNormalWidgets()
+	}
+	tc.termuiConsoleWidgets.PrepareListWithLogsForDisplay(tc.logLines)
+
+	ui.Clear()
+	ui.Render(tc.grid)
+}
+
+func (tc *TermuiConsole) prepareDataNormalWidgets() {
+	nonceI, _ := tc.termuiConsoleMetrics.Load(core.MetricNonce)
+	nonce := nonceI.(int)
+
+	synchronizedRoundI, _ := tc.termuiConsoleMetrics.Load(core.MetricSynchronizedRound)
+	synchronizedRound := synchronizedRoundI.(int)
+
+	currentRoundI, _ := tc.termuiConsoleMetrics.Load(core.MetricCurrentRound)
+	currentRound := currentRoundI.(int)
+
+	isSyncingI, _ := tc.termuiConsoleMetrics.Load(core.MetricIsSyncing)
+	isSyncing := isSyncingI.(int)
+
+	tc.termuiConsoleWidgets.PrepareSyncInfoForDisplay(nonce, currentRound, synchronizedRound, isSyncing)
+
+	publicKeyI, _ := tc.termuiConsoleMetrics.Load(core.MetricPublicKey)
+	publicKey := publicKeyI.(string)
+	tc.termuiConsoleWidgets.PreparePublicKeyForDisplay(publicKey)
+
+	shardIdI, _ := tc.termuiConsoleMetrics.Load(core.MetricShardId)
+	shardId := shardIdI.(int)
+	tc.termuiConsoleWidgets.PrepareShardIdForDisplay(shardId)
+
+	numConnectedPeersI, _ := tc.termuiConsoleMetrics.Load(core.MetricNumConnectedPeers)
+	numConnectedPeers := numConnectedPeersI.(int)
+
+	txPoolLoadI, _ := tc.termuiConsoleMetrics.Load(core.MetricTxPoolLoad)
+	txPoolLoad := txPoolLoadI.(int)
+	tc.termuiConsoleWidgets.PrepareTxPoolLoadForDisplay(txPoolLoad)
+	tc.termuiConsoleWidgets.PrepareNumConnectedPeersForDisplay(numConnectedPeers)
+
+	countConsensusI, _ := tc.termuiConsoleMetrics.Load(core.MetricCountConsensus)
+	countConsensus := countConsensusI.(int)
+
+	countLeaderI, _ := tc.termuiConsoleMetrics.Load(core.MetricCountLeader)
+	countLeader := countLeaderI.(int)
+
+	countAcceptedBlocksI, _ := tc.termuiConsoleMetrics.Load(core.MetricCountAcceptedBlocks)
+	countAcceptedBlocks := countAcceptedBlocksI.(int)
+
+	tc.termuiConsoleWidgets.PrepareConsensusInformationsForDisplay(countConsensus, countLeader, countAcceptedBlocks)
+}
+
+func (tc *TermuiConsole) changeConsoleDisplay() *ui.Grid {
+	if tc.numLogLines == 10 {
```

Comment:
Magic numbers here... put them in a variable.

Reply:
created const variables.